src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1

```diff
@@ -196,6 +197,31 @@ param(
     default { throw "7-Zip signalled an unknown error (code $exitCode)" }
   }
 
-  $env:ChocolateyPackageInstallLocation = $destination
-  return $destination
+  if ((test-path variable:url) -AND (@($url).count -gt 0))
+  {
+    # If our unzip resulted in a .tar file that matches our input zip file without it's extensions, unzip it
+    # does not work with archive names that contain a '.' before the extensions
+    $matchingtarname = "$destinationNoRedirection\$((split-path -leaf $url).split('.')[0]).tar"
+    Write-Debug "Checking to see if we have a flie called $matchingtarname"
+    If ((Test-path $matchingtarname) -AND (!$RecursiveCall))

```

Comment:
It is a recursivecall - it's just not unlimited recursion. 

I used a recursive call to take advantage of any of the things the function already does because I don't know what hard lessons have gone into creating that code.  For instance, the existing code pre-creates the target folder: https://github.com/chocolatey/choco/blob/47729b0e546a3e93f947903d72ee984a60c02e70/src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1#L105)

Recursion also means that any improvements to the code in the function don't have to be duplicated in the sub-unzip code.

Based on my limited, past exposure to the issue of nested archives - I was thinking that most cases could be handled with one recursion.  I also don't know the situations under which multiple recursion levels might cause problems - so thought it best not to leave the code open to problems.